diff --git a/serviceappfront/server/routes/Services.js b/serviceappfront/server/routes/Services.js
index 44f7abe7..a7688063 100644
--- a/serviceappfront/server/routes/Services.js
+++ b/serviceappfront/server/routes/Services.js
@@ -97,6 +97,6 @@ router.post("/updateProviouswork", auth, isBusiness,updateProviousWork);
 
 router.post("/deleteProviouswork",auth, isUser, deleteProviousWork);
 
-router.post("/getProviouswork",searchProviousWork);
+router.post("/getProviouswork",auth, isBusiness, searchProviousWork);
 
 module.exports = router
\ No newline at end of file
diff --git a/serviceappfront/src/component/Profile/ProviousWork.js b/serviceappfront/src/component/Profile/ProviousWork.js
index 038bd985..2a7c78b9 100644
--- a/serviceappfront/src/component/Profile/ProviousWork.js
+++ b/serviceappfront/src/component/Profile/ProviousWork.js
@@ -1,31 +1,24 @@
 import React, { useState } from "react";
-import {  useEffect } from "react";
+import { useSelector, useEffect } from "react";
 import { getProviousWork } from "../../services/oprerations/serviceAPIs";
-import { useSelector } from 'react-redux';
-
 const ProviousWork = ({ serviceId }) => {
   const { token } = useSelector((state) => state.auth);
 
-  // const { loading: AuthLoading } = useSelector((state) => state.auth);
+  const { loading: AuthLoading } = useSelector((state) => state.auth);
   const [loading, setLoading] = useState(null);
   const [data, setData] = useState([]);
   useEffect(() => {
-    const fun = async () => {
-      setLoading(true);
+    const setData = async () => {
       var data = null;
       if (serviceId)
         data = await getProviousWork({ service: serviceId }, token);
-      console.log(data);
-      setLoading(false)
-      return data
-
+      setData(data);
     };
-    const res=fun();
-    setData(res)
+    setData();
   }, []);
-  if (loading  || data.length===0) return <div className="spinner"></div>;
+  if (loading || AuthLoading || !data) return <div></div>;
   return (
-    <div className="p-2 mx-3 border-2 border-gray-200 mb-6">
+    <div>
       {/*  
 
 
@@ -63,7 +56,7 @@ const ProviousWork = ({ serviceId }) => {
 
 
 
-PROvious Work
+
 
       {/* add code here */}
     </div>
